python/az/aro/azext_aro/_aad.py

```diff
@@ -54,6 +56,17 @@ def get_service_principal(self, app_id):
         return None
 
     def create_service_principal(self, app_id):
-        return self.client.service_principals.create(ServicePrincipalCreateParameters(
-            app_id=app_id,
-        ))
+        service_principal = None
+        retries = 0
+        while True:
+            try:
+                service_principal = self.client.service_principals.create(
+                    ServicePrincipalCreateParameters(app_id=app_id))
+                break
```

Comment:
```suggestion
                return self.client.service_principals.create(
                    ServicePrincipalCreateParameters(app_id=app_id))
```

cleans up a little bit

Reply:
I thought about doing it that way - you're right, it's neater